File: app/src/main/java/ru/magflayer/spectrum/data/entity/converter/ColorPhotoConverter.java
package ru.magflayer.spectrum.data.entity.converter;

import ru.magflayer.spectrum.data.entity.ColorPhoto;
import ru.magflayer.spectrum.domain.entity.ColorPhotoEntity;

public class ColorPhotoConverter {

    public ColorPhoto convertToColorPhoto(final ColorPhotoEntity entity) {
        return new ColorPhoto(entity.getMillis(), entity.getFilePath(), entity.getRgbColors(),
                entity.getType().ordinal());
    }

    public ColorPhotoEntity convertToColorPhotoEntity(final ColorPhoto colorPhoto) {
        ColorPhotoEntity.Type type = ColorPhotoEntity.Type.values()[colorPhoto.getType()];
        return new ColorPhotoEntity(type, colorPhoto.getFilePath(), colorPhoto.getRgbColors(),
                colorPhoto.getId());
    }

}

File: app/src/main/java/ru/magflayer/spectrum/data/entity/converter/ColorPhotoConverter.kt
package ru.magflayer.spectrum.data.entity.converter

import ru.magflayer.spectrum.data.entity.ColorPhoto
import ru.magflayer.spectrum.domain.entity.ColorPhotoEntity

class ColorPhotoConverter : BaseConverter<ColorPhoto, ColorPhotoEntity>() {

    override fun convertToEntity(dto: ColorPhoto): ColorPhotoEntity {
        val type = ColorPhotoEntity.Type.values()[dto.type]
        return ColorPhotoEntity(type, dto.filePath, dto.rgbColors, dto.id)
    }

    override fun convertToDto(entity: ColorPhotoEntity): ColorPhoto {
        return ColorPhoto(
            entity.millis,
            entity.filePath,
            entity.rgbColors,
            entity.type.ordinal,
        )
    }
}
